### PR TITLE
[CC-19402] SR testing using RestApp

### PR DESCRIPTION
**Pull request by sudeshwasnik**
## Problem
Testing JSON_SR, PROTOBUF, AVRO formats with embedded schema-registry-cluster

##### Does this solution apply anywhere else?
- [x] yes
- [ ] no

##### If yes, where?
All connectors which support schema-based data

##### Testing done:
- [x] Unit tests
- [x] Integration tests
- [ ] System tests
- [ ] Manual tests - not required.

### CODE REVIEW

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorDataFormatIT.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+  protected void stopSchemaRegistry() throws Exception {
+    restApp.stop();
+  }
+
+  protected void waitForSchemaRegistryToStart() throws InterruptedException {
+    TestUtils.waitForCondition(
+        () -> restApp.restServer.isRunning(),
+        CONNECTOR_STARTUP_DURATION_MS,
+        "Connector tasks did not start in time."
```
Comment (sudeshwasnik) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorDataFormatIT.java:56]:
```suggestion
        "Schema-registry server did not start in time."
```

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorBaseIT.java [hunk 2]
```diff
@@ -28,22 +45,7 @@
...
-import java.util.Map;
-import org.apache.kafka.connect.json.JsonConverter;
-import org.apache.kafka.connect.storage.StringConverter;
-import org.apache.kafka.test.TestUtils;
-import org.junit.After;
-import org.junit.AfterClass;
-import org.junit.Before;
-
-public class ElasticsearchConnectorBaseIT extends BaseConnectorIT {
+public abstract class ElasticsearchConnectorBaseIT extends BaseConnectorIT {
```
Comment (sudeshwasnik) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorBaseIT.java:48]:
```suggestion
public class ElasticsearchConnectorBaseIT extends BaseConnectorIT {
```
unintended change.

Comment (sp-gupta) [reply src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorBaseIT.java:48]: @sudeshwasnik Are we committing this change?

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorDataFormatIT.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    converter.configure(Collections.singletonMap(
+            "schema.registry.url", restApp.restServer.getURI().toString()
+        ), false
+    );
+
+    // start the connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for schema-registry to spin up
+    waitForSchemaRegistryToStart();
```
Comment (amitr17) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorDataFormatIT.java:121]: already used at line 60.

Comment (sudeshwasnik) [reply src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorDataFormatIT.java:121]: gotcha. will remove 👍

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorDataFormatIT.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+@RunWith(Parameterized.class)
+public class ElasticsearchConnectorDataFormatIT extends ElasticsearchConnectorBaseIT {
+
+  protected void startSchemaRegistry() throws Exception {
+    int port = findAvailableOpenPort();
+    restApp = new RestApp(port, null, connect.kafka().bootstrapServers(),
+        KAFKASTORE_TOPIC, CompatibilityLevel.NONE.name, true, new Properties());
+    restApp.start();
+    waitForSchemaRegistryToStart();
```
Comment (sudeshwasnik) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorDataFormatIT.java:60]:
```suggestion
```

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorDataFormatIT.java (hunk 3 above)
Comment (sudeshwasnik) [reply src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorDataFormatIT.java:120]: done! removed !

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorBaseIT.java [hunk 5]
```diff
@@ -15,6 +15,23 @@
 
 package io.confluent.connect.elasticsearch.integration;
 
+import com.google.gson.JsonArray;
```
Comment (pbadani) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorBaseIT.java:18]: change in the codestyle?

Comment (sudeshwasnik) [reply src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorBaseIT.java:18]: hey @pbadani i didn't get you. I did not change the codestyle, any specific line you think is ill-formatted?

Comment (pbadani) [reply src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorBaseIT.java:18]: the imports were reorganized, nothing else.

**Review by pbadani — APPROVED** [review]

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorBaseIT.java (hunk 5 above)
Comment (sudeshwasnik) [reply src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorBaseIT.java:18]: oh, gotcha. I'll revert those back 👍

**Pull request merged**